src/vehicle.cpp

```diff
@@ -2360,6 +2365,96 @@ void Vehicle::HandleLoading(bool mode)
 	this->IncrementImplicitOrderIndex();
 }
 
+/**
+ * Checks whether a vehicle is waiting for automatic separation (if not,
+ * it is ready to depart)
+ */
+bool Vehicle::IsWaitingForAutomaticSeparation() const {
+	TimerGameTick::Ticks now = TimerGameCalendar::date.base() * Ticks::DAY_TICKS + TimerGameCalendar::date_fract;
+	return this->AutomaticSeparationIsEnabled() && this->first_order_last_departure > now;
+};
+
+/**
+ * If enabled, calculates the departure time for this vehicle based on the
+ * automatic separation feature.
+ */
+void Vehicle::UpdateAutomaticSeparation()
+{
+	/* Check this feature is enabled on the vehicle's orders */
+	if (!this->AutomaticSeparationIsEnabled()) return;
+
+	/* Only perform the separation at the first manual order which can load/unload (saves on storage) */
```

Comment:
This is a really good feature for network planning, because the player can make the first order a large station where vehicles can wait out the correction period. 

It would be even better if (as a separate PR), you allowed vehicles to wait in depots. JGRPP has an implementation of this that you could look at. Then players could set a depot as the first order and avoid delaying other vehicles. 😃